memory/jemalloc_nodump_allocator.cc

```diff
@@ -90,7 +96,30 @@ size_t JemallocNodumpAllocator::UsableSize(void* p,
 
 void* JemallocNodumpAllocator::Allocate(size_t size) {
   int tcache_flag = GetThreadSpecificCache(size);
-  return mallocx(size, MALLOCX_ARENA(arena_index_) | tcache_flag);
+
+  if (arena_indexes_.size() == 1) {
+    return mallocx(size, MALLOCX_ARENA(arena_indexes_[0]) | tcache_flag);
+  }
+
+  // `size` is used as a source of entropy to initialize each thread's arena
+  // selection.
```

Comment:
Are you sure this is a good source of entropy? Allocation requests may come from a small set of sizes. And why do we re-seed the generator every time, instead of having a TL generator? I think it would be much simpler to do something this:
```
static std::atomic<uint32_t> next_seed = 0;
thread_local Random tl_random{next_seed++};

// Use tl_random
```

This is a much more common pattern. Depending on the quality of `Random`, you may want to mix the seed.

Reply:
I seriously doubt it would make a difference. Re-seeding is more likely to be slower (some RNGs do pre-processing of the seed, though in the case of `rocksdb::Random` that's minimal). I'd land a counter-intuitive micro-optimization only if there is some supporting empirical evidence, and I don't think it's worth it here.

Reply:
Done, I changed it to `thread_local Random`.